pager/src/main/java/com/google/accompanist/pager/PagerState.kt

```diff
@@ -322,39 +335,60 @@ class PagerState(
                     )
                 }
 
+                // Keep track of velocity
+                lastVelocity = velocity
+
+                // Now scroll..
                 val coerced = value.coerceIn(0f, pageSize.toFloat())
-                dragBy((currentPageOffset * pageSize) - coerced)
+                scrollBy(coerced - (currentPageOffset * pageSize))
 
-                val pastLeftBound = initialVelocity > 0 &&
+                // If we've scroll our target page (or beyond it), cancel the animation
+                val pastStartBound = initialVelocity < 0 &&
                     (currentPage < targetPage || (currentPage == targetPage && currentPageOffset == 0f))
-
-                val pastRightBound = initialVelocity < 0 &&
+                val pastEndBound = initialVelocity > 0 &&
                     (currentPage > targetPage || (currentPage == targetPage && currentPageOffset > 0f))
 
-                if (pastLeftBound || pastRightBound) {
+                if (pastStartBound || pastEndBound) {
                     // If we reach the bounds of the allowed offset, cancel the animation
                     cancelAnimation()
-
                     currentPage = targetPage
                     currentPageOffset = 0f
                 }
             }
         } else {
             // Otherwise we animate to the next item, or spring-back depending on the offset
+            val targetPosition = currentPage + determineSpringBackOffset(
+                velocity = initialVelocity,
+                offset = targetOffset
+            )
             animate(
-                initialValue = currentPageOffset * pageSize,
-                targetValue = pageSize * determineSpringBackOffset(
-                    velocity = initialVelocity * -1,
-                    offset = targetOffset
-                ),
+                initialValue = globalPosition * pageSize,
+                targetValue = targetPosition * pageSize,
                 initialVelocity = initialVelocity,
                 animationSpec = spring()
```

Comment:
I know not touched in this PR but is the `spring` what we want for settling and/or should it be configurable? I feel like ViewPager had a specific interpolator, do we want to approximate that?

Reply:
Good call, done. I don't really want to reimplement VP's scroll interpolator, but I increased the spring stiffness which feels close.